.github/workflows/test.yml

```diff
@@ -30,6 +30,7 @@ jobs:
     - name: Test with Maven
       run: mvn -B verify --file pom.xml
     - name: Add coverage to PR
+      if: github.actor != 'dependabot[bot]'
```

Comment:
According to the [GitHub documentation](https://docs.github.com/en/code-security/code-scanning/troubleshooting-code-scanning/resource-not-accessible), this error may also be caused by not having the `pull_request` attribute, but it looks like we have it (at the top of this file). It has no attributes, but that should be fine. In the frontend repo, we have it like this, and we still have the same issue:

```yaml
on:
  push:
    branches: [main]
  pull_request:
    branches: [main]
```

It is also worth updating this particular action to its latest version (`madrapps/jacoco-report@v1.7.1`), although I doubt it fixes it.

The aforementioned documentation says:
> code scanning always allows the uploading of results when the pull_request event triggers the action run.

So I don't understand why we're seeing this error. According to [this Stackoverflow post](https://stackoverflow.com/questions/70435286/resource-not-accessible-by-integration-on-github-post-repos-owner-repo-ac), we could also define the job permissions like this:

```yaml
    permissions:
      pull-requests: write
```

So that may be worth a try as well.

In any case, the dependabot PRs shouldn't change the code coverage, so we can skip this step, as per your solution :-) It's just strange that this started failing all of a sudden.